Data/Sys/Shaders/default_pre_post_process.glsl

```diff
@@ -293,6 +293,24 @@ float4 AreaSampling(float3 uvw, float gamma)
 	return avg_color / (area_corners + area_edges + area_center);
 }
 
+/***** 4x Rotated Grid SuperSampling *****/
+
+float4 RGSS(float3 uvw, float gamma)
```

Comment:
Is this the main reference `https://blog.demofox.org/2015/04/23/4-rook-antialiasing-rgss/`? Please add it to the code if so.
By looking at this, I understand why this resampler would help hide aliasing on diagonal lines, but wouldn't it completely kill the quality otherwise? Out of 16 samples, you only account for 4, and none of these are close to the center. If you were upscaling, I would expect this to lose a lot of detail, and same if you were downscaling from a resolution that isn't at least 4 times bigger? Not sure though.

Reply:
I didn't recognize that until I checked the linked shadertoy, definitely. 

As for the idea, it's not sampling texels, it's sampling at positions inside the destination pixel. So if the source is equal or smaller than destination, you would end up sampling around the same data and there would be no change. For downsampling, it's a pretty common pattern that works. [nVIDIA MFAA](https://www.nvidia.com/en-us/geforce/news/multi-frame-sampled-anti-aliasing-delivers-better-performance-and-superior-image-quality/), see the Constant 4x pattern.

When I was messing around with filters for [PCSX2](https://github.com/PCSX2/pcsx2/pull/7647), this video was my main source: [Supersampling for Texture Maps](https://www.youtube.com/watch?v=naLIpFiH1sY). The Jittered Stochastic SuperSampling functions were quite fancy.

Reply:
Ok I understand now, seems like it makes sense there.
Just so you know, if you are interested, it would also be cool to implement this resampler:
https://github.com/BreadFish64/ScaleFish/tree/master
Taggin @Sam-Belliveau here as well.

Reply:
I tried its webgl demo and it kinda gives weird results, edges get lighter as well. [Test](https://imgur.com/a/jJP84N0)